python/src/aiconfig/editor/client/src/components/EditorContainer.tsx

```diff
@@ -262,9 +309,11 @@ export default function EditorContainer({
               <PromptContainer
                 index={i}
                 prompt={prompt}
+                getModels={callbacks.getModels}
                 onChangePromptInput={onChangePromptInput}
                 onChangePromptName={onChangePromptName}
                 onRunPrompt={onRunPrompt}
+                onUpdateModel={onUpdatePromptModel}
                 onUpdateModelSettings={onUpdatePromptModelSettings}
```

Comment:
Oh ok cool, so we are keeping them as separate actions?

Reply:
Ya, we should keep them separate since there's different logic for them:
- update settings --> keep same model name but new model settings object
- update model --> change model name and either keep settings or clear them